### DISCUSSION TITLE
Merging Dials from different divices

### DISCUSSION THREAD

**Discussion by volvisti** [fastaddons/GroupSpeedDial#140] [Q&A] [answered]
Hi there,
I'm wondering how to merge the dials I created on two different devices.

I just saved the dials of both devices locally. 
Loading the dials from the one to the other just overwrites the dials on the second one.
Bad idea, but thanks to the programmer there is a history you can restore the old dials from. 

Synchronzied the one (leading system) to the cloud.
Now I have clouddials. That's nice, but how can I get locally saved dials to the clouddials?
The other way round is easy. (clouddial to local system) :-S

Thanks for helping me.

Have a nice day 
S.Roggatz

**Comment by fastaddons** [comment]
Hello,
I should implement this since this has been asked before a few times.
Here is a manual way to import dials from a backup file while keeping existing dials:

1) export your dials (you want to import to the other account) to a file - from the Options page / Import backup tab.

2) open dials page on the PC where you want to append those exported dials - not the cloud dials page, normal dials page you see when you open new tab page

3) now the tricky part - open developer tools by pressing "Ctrl + Shift + K"

4) paste there this code and press Enter:
```
(() => {
  document.body.textContent = '';
  document.body.appendChild(Object.assign(document.createElement('input'), {style: 'position: fixed; left: 50%; top: 50%;', type: 'file', accept: '.group_speed_dial', oninput: e => {
      const reader = new FileReader();
      reader.onload = async e => {
        const groups = JSON.parse(reader.result).groups;
        const currentGroups = (await browser.storage.local.get()).groups;
        const maxId = Math.max(...currentGroups.map(g => g.id)) + 1;
        groups.forEach((g, i) => {
          g.id = maxId + i;
          currentGroups.push(g);
        });
        browser.storage.local.set({groups: currentGroups});
      };
      reader['readAsText'](e.target.files[0]);
    }}))
})()
```
5) in the middle of the screen you will see a small input button - click it and load the exported backup file.

That's it. Then just reload the page or open a new tab and voilà, your groups/dials are now there.

As always, if anything breaks, you can undo any change using the History feature from the main menu.

I hope this helps :)

**Reply by ladave69** [reply]
Have you implemented a way to do this Now?

**Reply by ladave69** [reply]
This is exactly the problem I have.

**Reply by Johnny531965** [reply]
I will try that soon

**Comment by ladave69** [comment]
I got it done somehow. What I did was change the name of the 4 key groups that I was interested in so they wouldn't overlap with what I had on the 2nd computer and hit synchronization and it transferred them over.

**Comment by Juraj-Masiar** [comment, marked as answer]
@ladave69 yes, this has been implemented already - you can export dials from one device and then while importing them to other device, simply uncheck the "Replace existing dials" checkbox:
![image](https://github.com/fastaddons/GroupSpeedDial/assets/6358971/61b71639-a821-466d-8dc8-3d45db958b09)

This will append imported groups after your current groups.

**Comment by ladave69** [comment]
Thank you!!😄
